src/shared/ui/modal/Modal.tsx

```diff
@@ -0,0 +1,48 @@
+import React, { useRef } from 'react';
+import { useModalStore } from '@/shared/stores/modalStore';
+import { Button } from '@/shared/ui/button/Button';
+import {
+  ButtonWrapper,
+  Content,
+  ModalBackground,
+  ModalContainer,
+} from '@/shared/ui/modal/Modal.style';
+
+export const Modal = () => {
+  const { isModalOpen, handleModal, content, type, onConfirm } = useModalStore();
+  const modalBackground: React.MutableRefObject<HTMLDivElement | null> = useRef(null);
```

Comment:
이렇게 표현해주는게 적절할 것 같네요~
```suggestion
  const modalBackground  = useRef<HTMLDivElement>(null);
```